src/app/manage/[clubId]/create/CreateEventForm.tsx

```diff
@@ -0,0 +1,130 @@
+'use client'
+
+import { useEffect, useState } from "react";
+import { type SelectClub } from "@src/server/db/models";
+import { createEventSchema } from "@src/utils/formSchemas";
+import { useForm } from "react-hook-form";
+import { api } from "@src/trpc/react";
+import { type z } from "zod";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useRouter } from "next/navigation";
+import { UploadIcon } from "@src/icons/Icons";
+import EventCardPreview from "./EventCardPreview";
+import TimeSelect from "./TimeSelect";
+import { type RouterOutputs } from "@src/trpc/shared";
+
+const CreateEventForm = ({ clubId, officerClubs }: { clubId: string, officerClubs: SelectClub[]}) => {
+	const {
+		register,
+		handleSubmit,
+		watch,
+		setValue,
+		getValues,
+	} = useForm<z.infer<typeof createEventSchema>>({
+		resolver: zodResolver(createEventSchema),
+		defaultValues: {
+			clubId: clubId,
+		},
+		mode: "onSubmit",
+	});
+	const router = useRouter();
+	const [watchDescription, watchStartTime] = watch(['description', 'startTime']);
+	const [eventPreview, setEventPreview] = useState<RouterOutputs['event']['findByFilters']['events'][number]>({
+		name: "",
+		clubId,
+		description: "",
+		location: "",
+		liked: false,
+		id: "",
+		startTime: new Date(Date.now()),
+		endTime: new Date(Date.now()),
+		club: officerClubs.filter((v) => v.id == clubId)[0]!,
+	});
+	useEffect(() => {
+		const subscription = watch((data, info) => {
+			const { name, clubId, description, location, startTime, endTime } = data;
+			const club = officerClubs.find((val) => val.id == data.clubId);
+			if (club) {
+				setEventPreview({
+					name: name || "",
+					clubId: clubId || "",
+					description: description || "",
+					location: location || "",
+					liked: false,
+					id: "",
+					startTime: startTime?.toString() === "" || startTime == undefined ? new Date(Date.now()) : new Date(startTime),
+					endTime: endTime?.toString() === "" || endTime?.toString() == "Invalid Date" || !endTime ? new Date(Date.now()) : new Date(endTime),
+					club,
+				});
+			}
+			if (info.name == "clubId") {
+				router.replace(`/manage/${data.clubId}/create`);
+			}
+		});
+		return () => subscription.unsubscribe();
+	}, [router, watch, officerClubs]);
+
+	const createMutation = api.event.create.useMutation({
+		onSuccess: () => { location.reload(); }
```

Comment:
On success either show a success message or redirect to the event page using `Router.push`